### ISSUE TITLE
Reading chunks from a large file fails

### ISSUE THREAD

**Issue by 7adidaz** [itinance/react-native-fs#1233]
I'm trying to read a chunk of size 50 MB from a ~130 MB file, I'm using [RNFS.read](https://github.com/itinance/react-native-fs?tab=readme-ov-file#readfilepath-string-length--0-position--0-encodingoroptions-any-promisestring), according to docs this reads a specific length starting from an offset, but this 50MB chunk tries to allocate memory equal to the whole file size and this fails due to ram size. 

Tried with 15 MB, and it worked like a charm. 
Tried with 30 MB, but it tried to allocate 80 MB of RAM which failed. 

Am I missing something here? why it does try to allocate more memory than it need?

**Comment by sagark1510** [comment]
Same problem. @7adidaz  did you find any solution?

**Comment by 7adidaz** [comment]
@sagark1510 nope, if I remember correctly, there was a memory leak related to this function, check out: [785](https://github.com/itinance/react-native-fs/issues/785)

**Comment by bibhuti9** [comment]
Any update?
